packages/ripple-binary-codec/src/types/st-object.ts

```diff
@@ -116,14 +116,25 @@ class STObject extends SerializedType {
       return Object.assign(acc, handled ?? { [key]: val })
     }, {})
 
-    let sorted = Object.keys(xAddressDecoded)
-      .map((f: string): FieldInstance => definitions.field[f] as FieldInstance)
-      .filter(
-        (f: FieldInstance): boolean =>
-          f !== undefined &&
-          xAddressDecoded[f.name] !== undefined &&
-          f.isSerialized,
+    function isValidFieldInstance(
+      f: FieldInstance | undefined,
+    ): f is FieldInstance {
+      return (
+        f !== undefined &&
+        xAddressDecoded[f.name] !== undefined &&
+        f.isSerialized
       )
+    }
+
+    let sorted = Object.keys(xAddressDecoded)
+      .map((f: string): FieldInstance | undefined => {
+        if (!(f in definitions.field)) {
+          if (f[0] === f[0].toLowerCase()) return undefined
```

Comment:
I don't understand this condition -- Can you point me to the documentation which explains the capitalization of the first character?

I observe examples where the first character of the field-name is both upper and lower case. Ex: https://github.com/XRPLF/xrpl.js/blob/effe7571f38827a7a44145e7163e1298ad2a262c/packages/ripple-binary-codec/src/enums/definitions.json#L44

Is there any explanation here? https://xrpl.org/docs/references/protocol/binary-format

Reply:
Lower-case fields are synthetic fields that aren't serialized, like `hash` or `date`. They're included in `tx` outputs, so if you're re-feeding that into the binary codec, you want those to be discarded instead of erroring.

Reply:
okay thanks. This is useful information, it will be helpful if its mentioned in the docs. Perhaps I missed it?